### PR TITLE
audio-sync

### PR DESCRIPTION
**Pull request by Fattimo**
## audio sync

adds an audio sync feature

### How to Test

1. go to ssr page and test it out
2. can upload own audio, or use sample never gonna give you up audio

### Important Changes

- N/A

### Related Github Issues

- #19 

### Related PRs

- #PRNUMBER

### Checklist

- [ ]  Database schema docs have been updated or are not necessary
- [ ]  Code follows design and style guidelines
- [ ]  Code is commented with doc blocks
- [ ]  Tests have been written and executed or are not necessary
- [ ]  Latest code has been rebased from base branch (usually `develop`)
- [ ]  Commits follow guidelines (concise, squashed, etc)
- [ ]  Github issues have been linked in relevant commits
- [ ]  Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### CODE REVIEW

**Review by cmurray191 — CHANGES_REQUESTED** [review]
Looks good so far. It doesn't seem to be working for me the way it was when you showed it to me. When I press begin syncing, nothing happens except that it says "Currently: Syncing" and the first word is highlighted. I am also getting an application error when I try to edit the text in the box. It happens when I backspace before the beginning of the word 'You' on the second line.

Also, I don't want to close out the issue for "Develop tools for syncing audio with text" until after all of the dependencies are addressed.

**Review by cmurray191 — APPROVED** [review]

**Review by cmurray191 — CHANGES_REQUESTED** [review]
Everything seems to be working properly, except for two things:
1. Begin syncing/stop syncing buttons should take the role of the play/pause audio buttons, because otherwise you have to click play and then click begin syncing and the audio doesn't end up syncing with when you press spacebar (you could even pause the audio but the timer for syncing continues). This is bound to cause confusion and improper syncing.

2. JSON mode does not work as desired. Right now, the JSON mode allows the user to copy/paste the json into the text box and it will use this to read the text. However, the goal is to be able to use the json to skip ahead to the most recent timestamp and corresponding place in the text to continue where they left off. This means that there should be the text from the book as well as a way to import json that captures the state of the previous session, allowing the user to begin syncing at the exact point they left off at.

tl;dr Changes:
1. Begin/stop syncing == play/pause audio
2. Export incomplete json at any time, import that json to begin where you left off

**Comment by Fattimo** [comment]
1. The timer for syncing is based on the actual time the audio is at. There is no timer for syncing, it's all based on the current timestamp of the audio file when the spacebar is pressed. I can add a feature that will autoplay the audio when begin synced is clicked, and another feature to automatically pause the audio file when stop syncing is clicked.

**Review by cmurray191 — APPROVED** [review]
looks good
